### PR TITLE
Added the ability to stop watching a repository

### PR DESCRIPTION
**Pull request by RuiAAPeres**

### CODE REVIEW

File: OctoKit/OCTClient+Watching.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@class OCTRepository;
+
+@interface OCTClient (Watching)
+
+/// Stops watching the repository.
+///
+/// repository      - The repository in which to stop watching. This must not be nil.
+//
+/// Returns a signal which will send complete, or error.
+
```
Comment (joshaber) [inline OctoKit/OCTClient+Watching.h]: Style: no new line here.

File: OctoKit/OCTClient+Watching.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#import "OCTClient.h"
+
+@class OCTRepository;
+
+@interface OCTClient (Watching)
+
+/// Stops watching the repository.
+///
+/// repository      - The repository in which to stop watching. This must not be nil.
```
Comment (joshaber) [inline OctoKit/OCTClient+Watching.h]: Style: don't need all the space before the `-`

File: OctoKit/OCTClient+Watching.m [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#import "OCTClient+Private.h"
+#import "OCTRepository.h"
+
+#import <ReactiveCocoa/ReactiveCocoa.h>
+#import "RACSignal+OCTClientAdditions.h"
+
+@implementation OCTClient (Watching)
+
+- (RACSignal *)stopWatchingRepository:(OCTRepository *)repository;
+{
```
Comment (joshaber) [inline OctoKit/OCTClient+Watching.m]: Style: opening `{` on the same line as the method name.

File: OctoKit/OCTClient+Watching.m [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	NSParameterAssert(repository != nil);
+	NSParameterAssert(repository.name.length > 0);
+	NSParameterAssert(repository.ownerLogin.length > 0);
+	
+	if (!self.authenticated) return [RACSignal error:self.class.authenticationRequiredError];
+	
+	NSURLRequest *request = [self requestWithMethod:@"DELETE" path:[NSString stringWithFormat:@"repos/%@/%@/subscription",repository.ownerLogin, repository.name] parameters:nil];
```
Comment (joshaber) [inline OctoKit/OCTClient+Watching.m]: Style: space between the comma and `repository.ownerLogin`.

File: OctoKit.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -64,6 +64,12 @@
 		9CB058A017933F51002CF498 /* OCTContentSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = 9CB0589F17933F51002CF498 /* OCTContentSpec.m */; };
 		9CF7FB9617957A6A00CA34FF /* repository.json in Resources */ = {isa = PBXBuildFile; fileRef = 9CF7FB9517957A6A00CA34FF /* repository.json */; };
 		9CF7FBBC179581EC00CA34FF /* user_starred.json in Resources */ = {isa = PBXBuildFile; fileRef = 9CF7FBBB179581EC00CA34FF /* user_starred.json */; };
+		C7C557DC1AB6348C00181B7E /* OCTClient+Watching.h in Headers */ = {isa = PBXBuildFile; fileRef = C7C557DA1AB6348C00181B7E /* OCTClient+Watching.h */; };
```
Comment (joshaber) [inline OctoKit.xcodeproj/project.pbxproj]: Are these headers public? It's hard to tell from the diff but they should be.

**Comment by joshaber** [comment]
Awesome :metal: 

Could you also add `OCTClient+Watching.h` to the framework header? :cocktail:

File: OctoKit.xcodeproj/project.pbxproj (hunk 5 above)
Comment (RuiAAPeres) [inline OctoKit.xcodeproj/project.pbxproj]: Do you mean adding them to the framework header?

Comment (joshaber) [inline OctoKit.xcodeproj/project.pbxproj]: Both :stuck_out_tongue: The header needs to be public so users can import it, and it should also be imported by default in the framework header.

**Comment by RuiAAPeres** [comment]
@joshaber The header was actually to the Project, an not Public.

**Comment by joshaber** [comment]
:cake: 

Just needs a merge now and it'll be good to go.

**Comment by RuiAAPeres** [comment]
@joshaber :+1:

**Comment by joshaber** [comment]
:boom:

**Pull request merged**